### PR TITLE
Resurrect internal release - use bintray for internal release

### PR DESCRIPTION
**Pull request by debasishg**
### What changes were proposed in this pull request?

* Resurrect `sbt internalRelease`. It was broken since we started releasing on to Sonatype. Internal release releases artifacts to bintray

### Why are the changes needed?

* Internal release is needed to test examples based on released artifacts. Every time it's not feasible to release on to sonatype - hence this releases stuff to lightbend's public bintray repository

### Does this PR introduce any user-facing change?

* Yes, adds an option to do an internal release of cloudflow libraries

### How was this patch tested?

* The patch has been tested manually by creating internal release

### CODE REVIEW

File: core/build.sbt [hunk 1]
```diff
@@ -443,7 +443,24 @@ def cloudflowModule(moduleID: String): Project = {
     .enablePlugins(AutomateHeaderPlugin)
 }
 
-lazy val commonSettings = Seq(
+// These settings are made active only when we use bintray for internal release
+// It is important that when we do final releases we need to invoke sbt as 
+// `sbt -Dsbt.sbtbintray=false`
```
Comment (skonto) [inline core/build.sbt]: We could have this set to false by default. Testing should be optionally switched on, its like a dry-run flag.

File: core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CommonSettingsAndTasksPlugin.scala [hunk 2]
```diff
@@ -42,11 +42,15 @@ object CommonSettingsAndTasksPlugin extends AutoPlugin {
   // common definitions
   final val CloudflowLocalConfigFile = ".lightbend/cloudflow/pipectl.json"
   final val CloudflowDockerBaseImage = "lightbend/cloudflow-base:1.3.0-M1-spark-2.4.4-flink-1.9.1-scala-2.12"
+  // used for internal release
+  final val CloudflowBintrayReleasesRepoUrl = "https://lightbend.bintray.com/cloudflow"
```
Comment (skonto) [inline core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CommonSettingsAndTasksPlugin.scala]: something like `CloudflowBintrayTestReleasesRepoUrl` could be more precise:

Comment (skonto) [reply core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CommonSettingsAndTasksPlugin.scala]: Is this going to be shipped with the plugin? I think this should be only enabled for internal use somehow.

File: core/build.sbt (hunk 1 above)
Comment (debasishg) [reply core/build.sbt]: @skonto The logic is we will use `internalRelease` much more frequently than the actual one. Hence the plugin is on by default and we set it to false _only during an actual release_. See the other PR on `cloudflow-release`

File: core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CommonSettingsAndTasksPlugin.scala (hunk 2 above)
Comment (debasishg) [reply core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CommonSettingsAndTasksPlugin.scala]: @skonto Strictly speaking it's not for test, it's for an internal release which, once done, all team members can use - hence the name.

Regarding shipping, why not ? The repository where we publish is a public one - hence people can use it from outside lightbend as well. And besides, how can we keep it only for internal use ?

File: core/build.sbt [hunk 3]
```diff
@@ -443,7 +443,24 @@ def cloudflowModule(moduleID: String): Project = {
...
+lazy val bintraySettings =
+  if (BintrayPlugin.isEnabledViaProp) {
+    Seq(bintrayOrganization := Some("lightbend"),
+      bintrayRepository := altBintrayRepository.value.getOrElse("cloudflow"),
+      bintrayOmitLicense := true,
+      publishMavenStyle := false,
+      resolvers += "Akka Snapshots" at "https://repo.akka.io/snapshots/",
```
Comment (olofwalker) [inline core/build.sbt]: Could combine this line and the line below.

Comment (debasishg) [reply core/build.sbt]: @olofwalker On a second look, I am thinking if we really need this akka snapshot repo added to the resolver ? Do we use snapshots ? /cc  @RayRoestenburg

File: core/build.sbt [hunk 4]
```diff
@@ -485,6 +502,11 @@ lazy val commonSettings = Seq(
   scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value,
 )
 
+val altBintrayRepository = settingKey[Option[String]]("Overrides the default bintray repository if defined")
+ThisBuild / altBintrayRepository := None
```
Comment (skyluc) [inline core/build.sbt]: Do we still use the `altBintrayRepository` setting ? It was used to push nightly builds to a different bintray repo, but we don't do this anymore.
I don't see a reference to it in the release scripts.

Comment (debasishg) [reply core/build.sbt]: yeah @skyluc  I think we can remove it and set the repo name straight to `cloudflow`

File: core/build.sbt (hunk 3 above)
Comment (debasishg) [reply core/build.sbt]: done ..

File: core/build.sbt (hunk 4 above)
Comment (debasishg) [reply core/build.sbt]: removed `altBintrayRepository` ..

**Pull request merged**